workflow/controller/controller.go

```diff
@@ -392,38 +395,34 @@ func (wfc *WorkflowController) initManagers(ctx context.Context) error {
 	return nil
 }
 
-func (wfc *WorkflowController) runConfigMapWatcher(stopCh <-chan struct{}) {
+func (wfc *WorkflowController) newConfigMapCtrlInformer() cache.SharedIndexInformer {
 	defer runtimeutil.HandleCrash(runtimeutil.PanicHandlers...)
 
 	ctx := context.Background()
-	retryWatcher, err := apiwatch.NewRetryWatcher("1", &cache.ListWatch{
-		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-			return wfc.kubeclientset.CoreV1().ConfigMaps(wfc.managedNamespace).Watch(ctx, metav1.ListOptions{})
-		},
+
+	informer := v1.NewConfigMapInformer(wfc.kubeclientset, wfc.GetManagedNamespace(), 20*time.Minute, cache.Indexers{
```

Comment:
`wfc.GetManagedNamespace()` is better than `wfc.managedNamespace`, but still not quite correct, per my above comment. I'm fine if this PR doesn't fix the other bug, so either we can leave as is or we need to modify this to correctly fix the other bug

Reply:
I'd vote for changing it to `wfc.namespace` (thanks for noticing that @agilgur5)

Reply:
Great, I'll update it.